app/Repositories/ProjectRepository.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Repositories;
+
+use App\Models\User;
+use App\Models\Project;
+use Illuminate\Pagination\LengthAwarePaginator;
+
+class ProjectRepository
```

Comment:
I have seen you have written test for project filtration in ProjectControllerTest, but the added benefit of repository is that we can have their dedicated tests in that we we don't have to do in controller. In controller we should be checking that logic that is inside them.